sources/kube.go

```diff
@@ -68,39 +66,17 @@ func (self *KubeSource) recordNodeError(name string) {
 	self.nodeErrors[name]++
 }
 
-func (self *KubeSource) addActivePodInstance(podName, podId, hostIp string) {
-	podInstance := PodInstance{Pod: podName, PodId: podId, HostIp: hostIp}
-	self.activePods[podInstance] = true
-}
-
-func (self *KubeSource) cleanupPods(pods []Pod) {
-	self.stateLock.Lock()
-	defer self.stateLock.Unlock()
-	newActivePods := make(map[PodInstance]bool)
-	newPodErrors := make(map[PodInstance]int)
-	for _, pod := range pods {
-		podInstance := PodInstance{Pod: pod.Name, PodId: pod.ID, HostIp: pod.HostIP}
-		if self.activePods[podInstance] {
-			newActivePods[podInstance] = true
-			if self.podErrors[podInstance] != 0 {
-				newPodErrors[podInstance] = self.podErrors[podInstance]
-			}
-		}
+func (self *KubeSource) recordPodError(pod Pod) {
+	// Heapster knows about pods before they are up and running on a node.
+	// Ignore errors for Pods that are not Running.
+	if pod.Status != "Running" {
+		return
```

Comment:
kubelet will set pod to running when at least one container instance is running. Will that return it back to the earlier state?